src/AspNetCore/Src/proxymodule.cxx

```diff
@@ -79,32 +79,82 @@ CProxyModule::OnExecuteRequestHandler(
     IHttpEventProvider *
 )
 {
-    m_pHandler = new FORWARDING_HANDLER(pHttpContext);
-    if (m_pHandler == NULL)
-    {
-        pHttpContext->GetResponse()->SetStatus(500, "Internal Server Error", 0, E_OUTOFMEMORY);
+    HRESULT hr;
+    APPLICATION_MANAGER* pApplicationManager;
+    APPLICATION* pApplication;
+    ASPNETCORE_CONFIG* config;
+    ASPNETCORE_APPLICATION* pAspNetCoreApplication;
+    ASPNETCORE_CONFIG::GetConfig(pHttpContext, &config);
+
+    // TODO store whether we are inproc or outofproc so we don't need to check the config everytime?
+    if (config->QueryHostingModel()->Equals(L"inprocess", true)) {
+        pApplicationManager = APPLICATION_MANAGER::GetInstance();
+        if (pApplicationManager == NULL)
+        {
+            hr = E_OUTOFMEMORY;
+            goto Failed;
+        }
+
+        hr = pApplicationManager->GetApplication(pHttpContext,
+            &pApplication);
+        if (FAILED(hr))
+        {
+            goto Failed;
+        }
+
+
+        hr = pApplication->GetAspNetCoreApplication(config, &pAspNetCoreApplication);
+        if (FAILED(hr))
+        {
+            goto Failed;
+        }
+
+        // Allow reading and writing to simultaneously
+        ((IHttpContext3*)pHttpContext)->EnableFullDuplex();
+
+        // Disable response buffering by default, we'll do a write behind buffering in managed code
+        ((IHttpResponse2*)pHttpContext->GetResponse())->DisableBuffering();
+
+        // TODO: Optimize sync completions
+        return pAspNetCoreApplication->ExecuteRequest(pHttpContext);
+    Failed:
+        pHttpContext->GetResponse()->SetStatus(500, "Internal Server Error", 0, E_APPLICATION_ACTIVATION_EXEC_FAILURE);
         return RQ_NOTIFICATION_FINISH_REQUEST;
     }
-
-    return m_pHandler->OnExecuteRequestHandler();
+    else
```

Comment:
this is too general. user may give random string now and later found behavior changed.  It's better check empty or specific string value with else if and then return error for other scenario